src/digital_ocean_client.py

```diff
@@ -0,0 +1,85 @@
+import boto3
+import pandas as pd
+from io import BytesIO
+from zipfile import ZipFile
+from io import StringIO
+import os
+from dotenv import load_dotenv
+import pdb
+import streamlit as st
+from datetime import datetime
+import shutil
+import geopandas as gpd
+
+load_dotenv()
+
+
+class DigitalOceanClient:
+    def __init__(self, bucket_name, repo_name):
+        self.bucket_name = bucket_name
+        self.repo_name = repo_name
+
+    def cache_key(self, url):
+        return str(
+            self.s3_resource()
+            .ObjectSummary(bucket_name=self.bucket_name, key=url)
+            .last_modified
+        )
+
+    def bucket(self):
+        return self.s3_resource().Bucket(self.bucket_name)
+
+    def get_folders(self):
+        return self.bucket().objects.filter(Prefix=f"{self.repo_name}/")
+
+    def unzip_csv(self, csv_filename, zipfile):
+        try:
+            with zipfile.open(csv_filename) as csv:
+                return pd.read_csv(csv, true_values=["t"], false_values=["f"])
+        except:
+            return None
+
+    def unzip_shapefile(self, table, zipfile):
+        try:
+            with zipfile as zf:
+                time = str(datetime.now().timestamp)
+                zf.extractall(path=f".library/{time}/{table}/")
+                gdf = gpd.read_file(f".library/{time}/{table}/{table}.shp")
+                shutil.rmtree(path=f".library/{time}")
+                return gdf
+        except:
+            return None
+
+    def s3_resource(self):
+        return boto3.resource(
+            "s3",
+            aws_access_key_id=os.getenv("AWS_ACCESS_KEY_ID"),
+            aws_secret_access_key=os.getenv("AWS_SECRET_ACCESS_KEY"),
+            endpoint_url=os.getenv("AWS_S3_ENDPOINT"),
+        )
+
+    @st.experimental_memo
+    def csv_from_DO(_self, url, cache_key, _kwargs={}):
```

Comment:
Will new `_kwargs` trigger an unnecessary re-query? Don't see the `_argument` you wrote about in the description